Tests/FeatherWidgets.TestIntegration/Navigation/LanguageSelectorWidgetTests.cs

```diff
@@ -141,6 +146,173 @@ public void LanguageSelectorWidget_CurrentLanguageNotIncluded()
+            var languageSelectorController = new LanguageSelectorController();
+            languageSelectorControl.Settings = new ControllerSettings(languageSelectorController);
+            languageSelectorController.Model.IncludeCurrentLanguage = true;
+
+            var controls = new List<System.Web.UI.Control>();
+            controls.Add(languageSelectorControl);
+
+            this.serverOperationsNews.CreateNewsItem("TestNewsItem");
+            var newsSelectorControl = new MvcControllerProxy();
+            newsSelectorControl.ControllerName = typeof(NewsController).FullName;
+            var newsController = new NewsController();
+            newsSelectorControl.Settings = new ControllerSettings(newsController);
+            controls.Add(newsSelectorControl);
+
+            var pageLanguages = new[]
+            {
+                this.sitefinityLanguages["English"],
+                this.sitefinityLanguages["Turkish"]
+            };
+
+            var pageName = "TestPage";
+
+            var createdPages = this.CreateLocalizedPage(pageName, pageLanguages);
+
+            // Add language selector widget to the en-US page
+            var currentPage = createdPages.First();
+            PageContentGenerator.AddControlsToPage(currentPage.Key, controls);
+
+            var items = newsController.Model.CreateListViewModel(null, 1).Items.ToArray();
+            var expectedDetailNews = (NewsItem)items[0].DataItem;
+
+            string url = UrlPath.ResolveAbsoluteUrl("~/" + pageName + currentPage.Value.Name);
+            string detailNewsUrl = url + expectedDetailNews.ItemDefaultUrl;
+
+            var pageContent = PageInvoker.ExecuteWebRequest(detailNewsUrl);
+            Assert.IsNotNull(pageContent);
+
+            var expectedLinkes = new Dictionary<string, string>()
+            {
+                { this.sitefinityLanguages["English"].NativeName, this.GetPageUrl(pageName, this.sitefinityLanguages["English"], true) },
+                { this.sitefinityLanguages["Turkish"].NativeName, this.GetPageUrl(pageName, this.sitefinityLanguages["Turkish"]) }
+            };
+
+            var notExpectedLinkes = new Dictionary<string, string>()
+            {
+                { this.sitefinityLanguages["Arabic"].NativeName, this.GetPageUrl(pageName, this.sitefinityLanguages["Arabic"]) },
+                { this.sitefinityLanguages["Serbian"].NativeName, this.GetPageUrl(pageName, this.sitefinityLanguages["Serbian"]) }
+            };
+
+            this.AssertLanguageLinks(pageContent, expectedLinkes, notExpectedLinkes);
+        }
+
+        [Test]
+        [Multilingual]
+        [Category(TestCategories.Navigation)]
+        [Author(FeatherTeams.Team7)]
+        [Description("Verifies language selector is included in detail view of content items")]
+        public void LanguageSelectorWidget_CurrentLanguageNotIncludedInDetailsViewOfContentItems()
```

Comment:
The test fails because on the detail page of the news item there is another link to comments. You have to change the AssertLanguageLinks method in order to work correctly.